lib/config/global.ts

```diff
@@ -1,40 +1,56 @@
 import type { RenovateConfig, RepoGlobalConfig } from './types';
 
-let repoGlobalConfig: RepoGlobalConfig = {};
-
-// TODO: once global config work is complete, add a test to make sure this list includes all options with globalOnly=true (#9603)
-const repoGlobalOptions = [
-  'allowCustomCrateRegistries',
-  'allowPlugins',
-  'allowPostUpgradeCommandTemplating',
-  'allowScripts',
-  'allowedPostUpgradeCommands',
-  'binarySource',
-  'customEnvVariables',
-  'dockerChildPrefix',
-  'dockerImagePrefix',
-  'dockerUser',
-  'dryRun',
-  'exposeAllEnv',
-  'migratePresets',
-  'privateKey',
-  'privateKeyOld',
-  'localDir',
-  'cacheDir',
-];
-
-export function setGlobalConfig(
-  config: RenovateConfig | RepoGlobalConfig = {}
-): RenovateConfig {
-  repoGlobalConfig = {};
-  const result = { ...config };
-  for (const option of repoGlobalOptions) {
-    repoGlobalConfig[option] = config[option];
-    delete result[option];
+export class GlobalConfig {
+  // TODO: once global config work is complete, add a test to make sure this list includes all options with globalOnly=true (#9603)
+  private static readonly OPTIONS = [
```

Comment:
I tried making this `(keyof RepoGlobalConfig)[]`, but had issues in `set` where `config[option]` was `unknown`.

It's probably not possible to do, but if you've got any ideas?